packages/azure-services/src/azure-batch/batch-task-config-generator.ts

```diff
@@ -12,14 +11,18 @@ export class BatchTaskConfigGenerator {
 
     public async getTaskConfig(taskId: string, messageText: string): Promise<BatchServiceModels.TaskAddParameter> {
         const commandLine = this.batchTaskCommandLineProvider.getCommandLine(messageText);
-        const maxTaskDurationInMinutes = (await this.batchTaskCommandLineProvider.getDefaultTaskConfig()).taskTimeoutInMinutes;
+        const batchTaskConfig = await this.batchTaskCommandLineProvider.getDefaultTaskConfig();
 
         return {
             id: taskId,
             commandLine: commandLine,
             resourceFiles: this.batchTaskCommandLineProvider.getResourceFiles(),
             environmentSettings: this.batchTaskCommandLineProvider.getAllEnvironmentSettings(messageText),
-            constraints: { maxWallClockTime: moment.duration({ minute: maxTaskDurationInMinutes }).toISOString() },
+            constraints: {
```

Comment:
I was wondering why are we adding this? because if some task failed because of timeout, should we increase the timeout too?

Reply:
We are decreasing time to keep task work files on a VM host from 7 to 3 days and adding retry to handle case when task was not able to finish normally ie task has crashed abnormally.